habitat/tasks/nav/nav.py

```diff
@@ -509,6 +509,41 @@ def update_metric(
         )
 
 
+@registry.register_measure
+class SoftSPL(SPL):
+    r"""Soft SPL
+
+    Similar to SPL with a relaxed soft-success criteria. Instead of a boolean
+    success is now calculated as 1 - (ratio of distance covered to target).
+    """
+
+    def _get_uuid(self, *args: Any, **kwargs: Any):
+        return "softspl"
+
+    def update_metric(self, episode, *args: Any, **kwargs: Any):
+        current_position = self._sim.get_agent_state().position.tolist()
+        distance_to_target = self._sim.geodesic_distance(
+            current_position, episode.goals[0].position
+        )
+
+        ep_soft_success = (
```

Comment:
Should this be `max(1 - distance_to_target / self._start_end_episode_distance, 0)`, that way the metric is in `[0, 1]`?